main/routines.c

```diff
@@ -545,10 +555,24 @@ extern const char *baseFilename (const char *const filePath)
 	 */
 	for (i = 0  ;  i < strlen (PathDelimiters)  ;  ++i)
 	{
+#ifdef HAVE_MBLEN
+		const char *p;
+		int ml;
+
+		for (p = filePath  ;  *p != '\0'  ;  ++p)
+		{
+			ml = mblen(p, MB_LEN_MAX);
+			if (ml > 1)
+				p += ml - 1;
+			else if (*p == PathDelimiters [i] && p > tail)
+				tail = p;
+		}
+#else
```

Comment:
why?  if you mean to support non-ASCII-compatible encodings, this seems incorrect to me.  If you don't, what's the point?
And anyway, if we really want to support on-ASCII-compatible encodings, we probably need a lot more work (including converting the input files content, as all parsers assume ASCII).

But note that of course, UTF-8 is ASCII-compatible.